R/msn_handlers.R

```diff
@@ -230,6 +230,13 @@ msn_constructor <-
     V(mst)$shape     <- "pie"
     V(mst)$pie       <- mlg.cp
     V(mst)$pie.color <- mlg.color    
+    # Transform pie made of single population into circle
+    pie.single <- sapply(V(mst)$pie, length) == 1
+    if (any(pie.single)) {
+      V(mst)$shape[pie.single] <- "circle"
+      V(mst)$color             <- rep(NA, length(V(mst)$name))
+      V(mst)$color[pie.single] <- unlist(V(mst)$pie.color[pie.single])
```

Comment:
This part unfortunately removes the names from the vector (for reasons that are beyond me). A way around this would be to save the circle colors as a temporary vector and then append the names separately like so:

```r
the_circles <- unlist(V(mst)$pie.color[pie.single])
V(mst)$color[pie.single] <- the_circles # set the color palette
names(V(mst)$color)[pie.single] <- names(the_circles) # set the population names
```